### PR TITLE
Refactor dialog

### PR DESCRIPTION
**Pull request by rppld**
# Description

This adds support for an optional prop `lockBodyScroll` that if `true` applies `overflow: hidden;` on the `<body>`. Also added a new export `useDialog()` – see the newly added story for how to use it.

### CODE REVIEW

File: src/components/Dialog/index.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { Component } from 'react'
+import React, { Component, useState, useEffect, useLayoutEffect } from 'react'
```
Comment (lucasfeliciano) [inline src/components/Dialog/index.js]: 😍

File: src/components/Dialog/index.js [hunk 2]
```diff
@@ -236,76 +250,112 @@ export class Dialog extends Component {
   }
 }
 
-export class DialogWindow extends Component {
-  static propTypes = {
-    on: PropTypes.bool.isRequired,
-    hide: PropTypes.func.isRequired,
-  }
+export function useDialog(props = {}) {
```
Comment (lucasfeliciano) [inline src/components/Dialog/index.js:244]: 😍

**Review by lucasfeliciano — APPROVED** [review]

**Comment by RobVermeer** [comment]
One issue, when opening the Dialog there is no padding added to the body, thus it can happen that the content moves when opening and closing the dialog.

**Review by RobVermeer — APPROVED** [review]
👍

**Pull request merged**